kms/auth-eth/contracts/KmsAuth.sol

```diff
@@ -120,6 +126,51 @@ contract KmsAuth is
         emit AppRegistered(appId);
     }
 
+    // Function to set AppAuth implementation contract address
+    function setAppAuthImplementation(address _implementation) external onlyOwner {
+        require(_implementation != address(0), "Invalid implementation address");
+        appAuthImplementation = _implementation;
+        emit AppAuthImplementationSet(_implementation);
+    }
+
+    // Factory method: Deploy and register AppAuth in single transaction
+    function deployAndRegisterApp(
+        address initialOwner,
+        bool disableUpgrades,
+        bool allowAnyDevice,
+        bytes32 initialDeviceId,
+        bytes32 initialComposeHash
+    ) external returns (address appId, address proxyAddress) {
+        require(appAuthImplementation != address(0), "AppAuth implementation not set");
+        require(initialOwner != address(0), "Invalid owner address");
+        
+        // Calculate app ID
+        appId = nextAppId();
+        
+        // Prepare initialization data
+        bytes memory initData = abi.encodeWithSelector(
+            bytes4(keccak256("initializeWithData(address,address,bool,bool,bytes32,bytes32)")),
+            initialOwner,
+            appId,
+            disableUpgrades,
+            allowAnyDevice,
+            initialDeviceId,
+            initialComposeHash
+        );
+        
+        // Deploy proxy contract
+        proxyAddress = address(new ERC1967Proxy(appAuthImplementation, initData));
+        
+        // Register to KMS
+        require(!apps[appId].isRegistered, "App already registered");
+        apps[appId].isRegistered = true;
+        apps[appId].controller = proxyAddress;
+        nextAppSequence[msg.sender]++;
+        
+        emit AppRegistered(appId);
```

Comment:
It looks like this part is a duplicate of the registerApp logic. Consider extracting the shared functionality into _registerApp.